api/core/v1alpha1/core.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package core.skv2.solo.io;
+
+option go_package = "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1alpha1/types";
+
+import "google/protobuf/wrappers.proto";
+
+import "gogoproto/gogo.proto";
+option (gogoproto.equal_all) = true;
+import "extproto/ext.proto";
+option (extproto.hash_all) = true;
+
+// Resource reference for an object
+message ObjectRef {
+
+    google.protobuf.StringValue api_group = 1;
+
+    google.protobuf.StringValue kind = 2;
+
+    string name = 3;
+
+    string namespace = 4;
+}
+
+// The State of a reconciled object
+enum State {
+    // Waiting to be processed.
+    PENDING = 0;
+
+    // Currently processing.
+    PROCESSING = 1;
+
+    // Invalid parameters supplied, will not continue.
+    INVALID = 2;
+
+    // Failed during processing.
+    FAILED = 3;
+
+    // Finished processing successfully.
+    ACCEPTED = 4;
```

Comment:
i'm not sure about making `State` a core api in skv2 yet. i think it's worth discussing at least